### PR TITLE
[docs] AI: Update AI documentation (MDL-83402)

### PR DESCRIPTION
**Pull request by davewoloszyn**
Documentation updates for MDL-83402. It has already been peer reviewed at this stage. Check tracker issue for more details.

### CODE REVIEW

**Comment by andrewnicols** [comment]
Note: This change must be backported to 4.5 docs as docs have now been forked.

**Comment by davewoloszyn** [comment]
I have added another commit for 4.5 dev docs

**Comment by davewoloszyn** [comment]
The related issue MDL-83402 has been reviewed and closed. This is ready for merging.

File: docs/apis/plugintypes/ai/provider.md [hunk 1]
```diff
@@ -5,26 +5,34 @@ tags:
...
 
 :::
 
+## Class implementation
+
 Providers are defined as classes in their own namespace according to their plugin name.
-The naming convention for Action classes is `aiprovider_<plugin name>`,
-for example: `aiprovider_openai`, `aiprovider_azureai`. With corresponding namespaces.
+The naming convention for a Provider class is `aiplacement_<plugin name>`.
+For example: `aiplacement_openai`, or `aiprovider_azureai` (with a corresponding namespace).
```
Comment (mickhawkins) [inline docs/apis/plugintypes/ai/provider.md:28]: There are two small typos here, referencing **aiplacement_** when it should be **aiprovider_**

**Review by mickhawkins — CHANGES_REQUESTED** [review]

File: docs/apis/plugintypes/ai/provider.md (hunk 1 above)
Comment (davewoloszyn) [reply docs/apis/plugintypes/ai/provider.md]: Thanks Mick. Good catch. I have fixed those up for 4.5 and 5.0

**Review by mickhawkins — APPROVED** [review]
Thanks for the update, this looks all good now.

**Pull request merged**